lib/devise/models/confirmable.rb

```diff
@@ -293,12 +299,8 @@ def send_confirmation_instructions(attributes={})
         # If the user is already confirmed, create an error for the user
         # Options must have the confirmation_token
         def confirm_by_token(confirmation_token)
-          original_token     = confirmation_token
-          confirmation_token = Devise.token_generator.digest(self, :confirmation_token, confirmation_token)
```

Comment:
We will need to try both tokens for a while because of backwards compatibility. So we will probably need to call  `find_or_initialize_with_error_by(:confirmation_token, confirmation_token)` twice, one with the token digest and another with the original token.